devito/finite_differences/finite_difference.py

```diff
@@ -366,3 +363,56 @@ def generate_fd_shortcuts(function):
             derivatives[name_fd] = (deriv, desciption)
 
     return derivatives
+
+
+def symbolic_weights(function, deriv_order, indices, dim):
+    return [function.coeff_symbol(indices[j], deriv_order, function, dim)
+            for j in range(0, len(indices))]
+
+
+def generate_indices(func, dim, diff, order, stagger=None, side=None):
+
+    # Check if called from first_derivative()
+    if bool(side):
+        if side == right:
+            ind = [(dim+i*diff) for i in range(-int(order/2)+1-(order % 2),
+                                               int((order+1)/2)+2-(order % 2))]
+        elif side == left:
+            ind = [(dim-i*diff) for i in range(-int(order/2)+1-(order % 2),
+                                               int((order+1)/2)+2-(order % 2))]
+        else:
+            ind = [(dim+i*diff) for i in range(-int(order/2),
+                                               int((order+1)/2)+1)]
+        x0 = None
+    else:
+        if func.is_Staggered:
+            if stagger == left:
+                off = -.5
+            elif stagger == right:
+                off = .5
+            else:
+                off = 0
+            ind = list(set([(dim + int(i+.5+off) * dim.spacing)
+                            for i in range(-order//2, order//2)]))
+            x0 = (dim + off*diff)
+            if order < 2:
+                ind = [dim + diff, dim] if stagger == right else [dim - diff, dim]
+
+        else:
+            ind = [(dim + i*dim.spacing) for i in range(-order//2, order//2 + 1)]
+            x0 = dim
+            if order < 2:
+                ind = [dim, dim + diff]
+    return ind, x0
+
+
+def form_side(dimensions, function):
```

Comment:
@mloubout @rhodrin 

future work: we can definitely store left/right/centered [here](https://github.com/opesci/devito/blob/master/devito/types/dense.py#L144) (and lines below) instead of 0/1/-1 ... and then just use `.val` each time `.staggered` is used (in a very few places, for example [here](https://github.com/opesci/devito/blob/master/devito/types/dense.py#L205) .

I'm fine to leave this code as is for this PR (it was around before so...), but bear in mind that this translation of information is not only useless but also a bit ugly